packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -187,23 +188,15 @@ export default abstract class SqlIntegration
     return false;
   }
 
-  isAutoGeneratingFactTablesSupported(): boolean {
-    const supportedEventTrackers: Record<AutoFactTableSchemas, true> = {
-      segment: true,
-      rudderstack: true,
-      ga4: true,
-      amplitude: true,
-    };
+  // Currently, if auto generating fact tables is supported, so is generating auto metrics
+  isAutoGeneratingMetricsSupported(): boolean {
+    return this.isAutoGeneratingFactTablesSupported();
+  }
 
-    if (
-      this.datasource.settings.schemaFormat &&
-      supportedEventTrackers[
-        this.datasource.settings.schemaFormat as AutoFactTableSchemas
-      ]
-    ) {
-      return true;
-    }
-    return false;
+  schemaFormatisAutoFactTablesSchemas(
+    schemaFormat: SchemaFormat
+  ): schemaFormat is AutoFactTableSchemas {
+    return supportedEventTrackers[schemaFormat as AutoFactTableSchemas];
```

Comment:
Nit. This return either true or undefined. Should make this return false instead of undefined.